### PR TITLE
Adds support for Planck Through Hole Kit

### PR DESCRIPTION
**Pull request by jackhumbert**
This also adds support for audio using D4/D5 on the Atmega32A.

### CODE REVIEW

File: keyboards/planck/thk/matrix.c [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+    // dip switch
+    dip_switch[0] = !(PIND & (1 << 0));
+    dip_switch[1] = !(PIND & (1 << 1));
+    dip_switch[2] = !(PIND & (1 << 4));
+    dip_switch[3] = !(PIND & (1 << 5));
+    for (uint8_t i = 0; i < 4; i++) {
+      if (last_dip_switch[i] ^ dip_switch[i])
+        dip_update(i, dip_switch[i]);
+        dip_update_kb(i, dip_switch[i]);
```
Comment (juno) [inline keyboards/planck/thk/matrix.c]: @jackhumbert 
I think these both two lines should be surrounded by if statement.

Like this:
```diff
diff --git a/keyboards/planck/thk/matrix.c b/keyboards/planck/thk/matrix.c
index b2028bfae..482861fbd 100644
--- a/keyboards/planck/thk/matrix.c
+++ b/keyboards/planck/thk/matrix.c
@@ -86,9 +86,10 @@ uint8_t matrix_scan(void) {
     dip_switch[2] = !(PIND & (1 << 4));
     dip_switch[3] = !(PIND & (1 << 5));
     for (uint8_t i = 0; i < 4; i++) {
-      if (last_dip_switch[i] ^ dip_switch[i])
+      if (last_dip_switch[i] ^ dip_switch[i]) {
         dip_update(i, dip_switch[i]);
         dip_update_kb(i, dip_switch[i]);
+      }
     }
     memcpy(last_dip_switch, dip_switch, sizeof(&dip_switch));
```

Anyway, I enjoying learn QMK codebase through my own Planck THK. Thank you!

Comment (jackhumbert) [reply keyboards/planck/thk/matrix.c]: Good catch! Thanks :)

File: keyboards/planck/thk/rules.mk [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+OPT_DEFS = -DDEBUG_LEVEL=0
+
+# custom matrix setup
+CUSTOM_MATRIX = yes
+SRC = matrix.c
+
+# programming options
+PROGRAM_CMD = ./util/atmega32a_program.py $(TARGET).hex
+
+LAYOUTS = ortho_4x12 planck_mit planck_grid
```
Comment (noroadsleft) [inline keyboards/planck/thk/rules.mk]: There is no planck_grid community layout as we seem to have settled on ortho_4x12 instead.

```suggestion
LAYOUTS = ortho_4x12 planck_mit
```

Comment (jackhumbert) [reply keyboards/planck/thk/rules.mk]: Thanks!

**Comment by drashna** [comment]
Anything other than the merge conflict holding this back from being merged?

**Comment by drashna** [comment]
@jackhumbert 

Merge conflict:

```git
<<<<<<< planck_thk
    #define readPin(pin) ((PIN_ADDRESS(pin, 0) & _BV(pin & 0xF)) != 0)
=======
    #define readPin(pin) ((bool)(PIN_ADDRESS(pin, 0) & _BV(pin & 0xF)))
>>>>>>> master
```

I wasn't sure which is correct/better, so ....

**Comment by drashna** [comment]
Aside from the merge conflict, this seems like it's ready to go. 

Travis fails on it, mostly because V-USB doesn't support NKRO, which is enabled on a lot of boards. 
But also because of community layouts.  Though, the default layouts compile fine.

**Comment by yanfali** [comment]
Still missing instructions on how to flash the board. Is it only ISP programmable at this time?

**Comment by drashna** [comment]
Woo merge conflicts.

**Comment by Sam-son** [comment]
Latest merge commit on this branch Breaks The Build. Last working commit I can find is 039129a. The bootloader also seems funky- or lacking docs? At any rate, building & flashing the combined bootloader/bootloadable (e.g. `make:production`) doesn't result in a functional keyboard.  

For anyone trying to build for the THK, try 
```
git checkout 039129ac
make clean;make planck/thk:default
```

**Comment by drashna** [comment]
So close, but so far.

**Pull request closed without merging**